infra/gravity/cluster_status.go

```diff
@@ -2,45 +2,108 @@ package gravity
 
 import (
 	"context"
-	"time"
 
+	"github.com/cenkalti/backoff"
+
+	"github.com/gravitational/robotest/lib/constants"
 	sshutils "github.com/gravitational/robotest/lib/ssh"
 	"github.com/gravitational/robotest/lib/utils"
 	"github.com/gravitational/robotest/lib/wait"
-
 	"github.com/gravitational/trace"
 )
 
-// Status walks around all nodes and checks whether they all feel OK
-func (c *TestContext) Status(nodes []Gravity) error {
-	c.Logger().WithField("nodes", Nodes(nodes)).Info("Check status on nodes.")
-	ctx, cancel := context.WithTimeout(c.ctx, c.timeouts.Status)
-	defer cancel()
+// returns nil if the Gravity Status is the expected status or an error otherwise
+type statusValidator func(s GravityStatus) error
 
-	retry := wait.Retryer{
-		Attempts: 100,
-		Delay:    time.Second * 20,
+// a negation / reimplementation of https://github.com/gravitational/gravity/blob/7.0.0/lib/status/status.go#L180-L185
+// I'd prefer isDegraded and negation later, but I didn't find an elegant way to
```

Comment:
Nit: To be honest I'm not sure about the value of comments like this, it reads more like a personal journal :) I would either remove this, or replace with a specific `TODO(walt)` item (which is the format we usually use for things like that).